### PR TITLE
adds emldown post

### PR DESCRIPTION
**Pull request by maelle**
As planned, `emldown` unconf blog post for August the 1st.

### CODE REVIEW

**Comment by maelle** [comment]
Note: I can only make changes until Thursday evening (8pm CEST), after that I'm off for a long week-end.

File: _posts/2017-08-01-emldown.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+title: "emldown: From machine readable EML metadata to a pretty documentation website"
```
Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: @maelle Thank you for contributing this post! The last of the unconf packages to be featured before we post "the Roads Not Taken". 

I'll include my comments to consider.

File: _posts/2017-08-01-emldown.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+author: "Maëlle Salmon, Andrew MacDonald, Kara Woo, Carl Boettiger, Jeff Hollister"
```
Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: Please see yaml in the checkers post for formatting (e.g. authors), use similar tags, etc:
https://github.com/ropensci/roweb/edit/master/_posts/2017-06-20-checkers.md

File: _posts/2017-08-01-emldown.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+date: "2017-08-01"
+output:
+  md_document:
+    variant: markdown_github
+    toc: true
+---
+
+We all know we should document our data, and some documentation is better than no documentation.  Even better, it's best practice to write metadata for a dataset based on a standard metadata format than in an unstructured document. A well-established standard is the [Ecological Metadata Language (EML)](https://knb.ecoinformatics.org/#external//emlparser/docs/index.html). If you follow the standard properly, you're sure to have documented your data well, which gives you a good feeling. Moreover, this will allow easier sharing of that dataset later.
```
Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: Add "rather" here: "...rather than in an unstructured document."

Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: Can you give just a little definition of what metadata is, with an example?

File: _posts/2017-08-01-emldown.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+---
+
+We all know we should document our data, and some documentation is better than no documentation.  Even better, it's best practice to write metadata for a dataset based on a standard metadata format than in an unstructured document. A well-established standard is the [Ecological Metadata Language (EML)](https://knb.ecoinformatics.org/#external//emlparser/docs/index.html). If you follow the standard properly, you're sure to have documented your data well, which gives you a good feeling. Moreover, this will allow easier sharing of that dataset later. 
+
+That said, it's not always fun to write standard metadata. Although there's a very good package for helping you create the EML, [rOpenSci's EML package](https://github.com/ropensci/eml), documenting the data can be quite tedious.  Furthermore, before you share the data on a public repository that enforces EML, the only prize you get is a happy conscience, which isn't very tangible. In our unconf project, we created immediate gratification for EML users: a package that transforms the non-human readable EML file into a pretty documentation website for any dataset! 
+
+# What's EML, exactly, and why can't a human read it?
+
```
Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: Maybe phrase this in the positive "everything she wants to document is present in the standard..." instead of "nothing she'd like to document is missing from the standard". As is, the phrase sounds awkward

File: _posts/2017-08-01-emldown.md (hunk 3 above)
Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: I see a bit of metadata definition below so perhaps review the order of sentences in the intro

File: _posts/2017-08-01-emldown.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+After creating EML documentation for your data, you get an XML document that's, well... not very human readable.
+
+![raw eml](figures/screenshot_raw_xml.png)
+
+(!!! function not working currently) In the `EML` package there's a function called `eml_view` relying on the [`listviewer` package](https://github.com/timelyportfolio/listviewer) to produce an interactive view of the XML in the Viewer pane of say, RStudio, which allows one to check some things quickly but which is far from being an user-friendly representation of the metadata.  Our goal with `emldown` was to bridge this gap and provide a quick an easy way to take EML and convert it to a more user-friendly web page.
```
Comment (stefaniebutland) [inline _posts/2017-08-01-emldown.md]: quick **and** easy

File: _posts/2017-08-01-emldown.md (hunk 1 above)
Comment (maelle) [inline _posts/2017-08-01-emldown.md]: corrected

File: _posts/2017-08-01-emldown.md (hunk 4 above)
Comment (maelle) [inline _posts/2017-08-01-emldown.md]: corrected

File: _posts/2017-08-01-emldown.md (hunk 5 above)
Comment (maelle) [inline _posts/2017-08-01-emldown.md]: done

File: _posts/2017-08-01-emldown.md (hunk 3 above)
Comment (maelle) [inline _posts/2017-08-01-emldown.md]: ok

Comment (maelle) [inline _posts/2017-08-01-emldown.md]: hope the rephrasing helps?

**Comment by maelle** [comment]
Thanks a lot for your feedback @stefaniebutland!

File: _posts/2017-08-01-emldown.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+title: "emldown: From machine readable EML metadata to a pretty documentation website"
+authors:
+  - name: Maëlle Salmon
+    url: http://www.masalmon.eu/
+  - name: Andrew MacDonald
+    url: http://aammd.info/
+  - name: Kara Woo
+    url: http://karawoo.com/
+  - name: Carl Boettiger
+    url: http://www.carlboettiger.info/
```
Comment (sckott) [inline _posts/2017-08-01-emldown.md]: @maelle can remove the url for Carl since our jekyll template links him to our `/staff` page

File: _posts/2017-08-01-emldown.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+---
+title: "emldown: From machine readable EML metadata to a pretty documentation website"
```
Comment (sckott) [inline _posts/2017-08-01-emldown.md]: @maelle can you do a dash instead of a `:` - jekyll doesn't like those in titles

File: _posts/2017-08-01-emldown.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+# What's EML, exactly, and why can't a human read it?
+
+[EML](https://en.wikipedia.org/wiki/Ecological_Metadata_Language) is a metadata standard originally created for the ecological sciences. In practice it's a set of XML schema documents, telling you what you need to document (e.g. the dataset creator, geographic coverage of the data, etc.) and how to build those documents (format of the XML). As mentioned previously, the [`EML` R package](https://ropensci.github.io/EML/) provides tools to create and read EML without needing to learn about XML thanks to helper functions and extensive documentation. Although its name contains the word _ecological_, one can use EML for documenting other datasets.  One of our team members uses it for documenting epidemiological datasets, because everything she wants to document is present in the standard, and because of the existence of the `EML` R package.
```
Comment (sckott) [inline _posts/2017-08-01-emldown.md]: > One of our team members uses it for documenting epidemiological datasets, because everything she wants to document is present in the standard, and because of the existence of the `EML` R package.

maybe reword, last part after comma sounds awkward

File: _posts/2017-08-01-emldown.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  - name: Jeff Hollister
+    url: https://jwhollister.com/
+date: "2017-08-01"
+output:
+  md_document:
+    variant: markdown_github
+    toc: true
+---
+
+We all know we should document our data, and some documentation is better than no documentation.  Even better, it's best practice to write metadata, i.e. structured documentation, for a dataset based on a standard machine-readable metadata format rather than in an unstructured document. A well-established standard is the [Ecological Metadata Language (EML)](https://knb.ecoinformatics.org/#external//emlparser/docs/index.html). If you follow the standard properly, you're sure to have documented your data well, which gives you a good feeling. Moreover, this will allow easier sharing of that dataset later.
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: "Even better, it's best practice" 

better and best in the same sentence reads a bit awkward.

File: _posts/2017-08-01-emldown.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+---
+
+We all know we should document our data, and some documentation is better than no documentation.  Even better, it's best practice to write metadata, i.e. structured documentation, for a dataset based on a standard machine-readable metadata format rather than in an unstructured document. A well-established standard is the [Ecological Metadata Language (EML)](https://knb.ecoinformatics.org/#external//emlparser/docs/index.html). If you follow the standard properly, you're sure to have documented your data well, which gives you a good feeling. Moreover, this will allow easier sharing of that dataset later. 
+
+That said, it's not always fun to write standard metadata. Although there's a very good package for helping you create the EML, [rOpenSci's EML package](https://github.com/ropensci/eml), documenting the data can be quite tedious.  Furthermore, before you share the data on a public repository that enforces EML, the only prize you get is a happy conscience, which isn't very tangible. In our unconf project, we created immediate gratification for EML users: a package that transforms the non-human readable EML file into a pretty documentation website for any dataset!
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: "If you follow the standard properly, you're sure to have documented your data well, which gives you a good feeling. " Could benefit from a rewrite.

Comment (karthik) [inline _posts/2017-08-01-emldown.md]: "That said, it's not always fun to write standard metadata." Should that be standardized?

Comment (karthik) [inline _posts/2017-08-01-emldown.md]: Link to unconf17.ropensci.org for the word unconf in _In our unconf project_

File: _posts/2017-08-01-emldown.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+That said, it's not always fun to write standard metadata. Although there's a very good package for helping you create the EML, [rOpenSci's EML package](https://github.com/ropensci/eml), documenting the data can be quite tedious.  Furthermore, before you share the data on a public repository that enforces EML, the only prize you get is a happy conscience, which isn't very tangible. In our unconf project, we created immediate gratification for EML users: a package that transforms the non-human readable EML file into a pretty documentation website for any dataset! 
+
+# What's EML, exactly, and why can't a human read it?
+
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: "What's EML, exactly, and why can't a human read it"

suggestion: why can't humans read it?

File: _posts/2017-08-01-emldown.md (hunk 8 above)
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: Suggestion: replace _created for the ecological sciences_ with _created for ecologists_

Comment (karthik) [inline _posts/2017-08-01-emldown.md]: "one can use EML for documenting other datasets"

What other datasets? All other datasets?

Would read better if it said "Can be used for datasets from any domain/field"

File: _posts/2017-08-01-emldown.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+After creating EML documentation for your data, you get an XML document that's, well... not very human readable.
+
+![raw eml](/assets/blog-images/2017-08-01-emldown/screenshot_raw_xml.png)
+
+(!!! function not working currently) In the `EML` package there's a function called `eml_view` relying on the [`listviewer` package](https://github.com/timelyportfolio/listviewer) to produce an interactive view of the XML in the Viewer pane of say, RStudio, which allows one to check some things quickly but which is far from being an user-friendly representation of the metadata.  Our goal with `emldown` was to bridge this gap and provide a quick and easy way to take EML and convert it to a more user-friendly web page.
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: Replace _far from being an user-friendly_ with _...a user friendly_

File: _posts/2017-08-01-emldown.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+# What does `emldown` do with your EML?
+
+After installing the package, currently from Github, you can apply it to your EML...
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: After installing the package, currently from Github

replace with

After installing the package from GitHub,...

File: _posts/2017-08-01-emldown.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+```r
+devtools::install_github("ropenscilabs/emldown")
+library("emldown")
+render_eml(path_to_my_eml)
+```
+
+and get something like this!
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: replace with: and you get something like this!

File: _posts/2017-08-01-emldown.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+![emldown](/assets/blog-images/2017-08-01-emldown/screenshot_emldown.png)
+
+This format is much more likely to make you and your collaborators happy. Note how little effort you needed to invest into making it! Viewing your metadata in this way makes it easy to read, and easy to spot any errors.
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: _This format is much more likely to make you and your collaborators happy._

Maybe say something like the format is more engaging, visual, easy to digest etc?

File: _posts/2017-08-01-emldown.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+Right now, we are able to capture some of the most common parts of Ecological Metadata Language, including the Title, Abstract, Authors, Keywords, Coverage (where in space and time the samples were taken), the Data Tables and Units associated with these. Over time we plan to add support for additional components of the EML specification.
+
+![demo3](/assets/blog-images/2017-08-01-emldown/emldown_demo3.gif)
+
+
+You can see a live example of a website created with `emldown` [here](http://aammd.info/emldown/test.html)
+
+# How can you contribute?
+
+Use the package to transform the EML you have lying around on your PC into a pretty website... and if you find a bug while doing so, we'll be happy to tackle it! Report any issue or feature request [here](https://github.com/ropenscilabs/emldown/issues), or feel free to contribute with code.
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: replace with: _pretty website and_

File: _posts/2017-08-01-emldown.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+We're very happy to have been able to create this working package in two days (and thankful for that opportunity, thanks rOpenSci!); and we not so secretly hope that it will contribute to making writing good metadata more attractive and therefore more common.
```
Comment (karthik) [inline _posts/2017-08-01-emldown.md]: Replace not so secretly with not-so-secretly

**Comment by maelle** [comment]
Thanks for your useful feedback @sckott @karthik ! I hope I haven't missed anything. I can do changes myself until today 11am Pacific time.

**Comment by karawoo** [comment]
If there's anything else after @maelle goes offline I can take over.

**Comment by karawoo** [comment]
Though I don't have push access to her fork, so might require some extra git finagling.

**Comment by maelle** [comment]
Added you as collaborator does it give you push access?

**Comment by karawoo** [comment]
Yup 👍

**Comment by karawoo** [comment]
I've added a little bit to the beginning of the post to try to give more context about EML and why it's useful. Though now it takes a bit longer before we get to emldown, which is the main topic of the post. Let me know if you have any feedback or want additional changes.

**Comment by karthik** [comment]
@karawoo The post looks great and ready to merge. One small edit (which I am also happy to do post merge if you just want to enjoy your weekend) is to remove the ellipsis at the end of _After installing the package from Github, you can apply it to your EML..._

**Comment by karawoo** [comment]
I can make that edit, should the ellipsis be replaced with anything else since the sentence continues after the code chunk?

**Comment by karthik** [comment]
@karawoo Ah, never mind. I didn't notice that. PR accepted! Thanks Kara and Maelle! 💯

**Comment by maelle** [comment]
Thanks rOpenSci team and thanks @karawoo for handling this!
